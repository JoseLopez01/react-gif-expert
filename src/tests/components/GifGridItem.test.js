import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('pruebas en GifGridItem', () => {

  
  const title = 'A title';
  const url = 'http://imagen.com';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Debe mostrar bien el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe tener un parrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('Debe tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Debe tener animate__fadeIn', () => {
    const div = wrapper.find('div');
    expect(div.hasClass('animate__fadeIn')).toBe(true)
  })

})