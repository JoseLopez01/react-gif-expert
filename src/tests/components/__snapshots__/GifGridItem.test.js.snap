// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem Debe mostrar bien el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="http://imagen.com"
  />
  <p>
    A title
  </p>
</div>
`;
